MAUI deep linking guidance

guardrex opened the pull request:
Fixes #28401

* I made a couple of guesses at wording for tech that I'm not very familiar with. Check carefully to correct any 💥 in meaning that I introduced 🙈.
* You could update the README of the sample app to just point to this article/section. The link is ...

  `https://learn.microsoft.com/aspnet/core/blazor/hybrid/routing?pivots=maui#deep-linking`

* Not sure what's going on with the original PU issue ... if this content just provides a workaround approach until the framework has formal API at .NET 9. Adjust the lead-in wording accordingly. If something is coming for .NET 9, I recommend leaving something in the opening remarks like what I have. Devs may schedule a revisit on deep linking in 2024 on that basis.

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/hybrid/routing.md](https://github.com/dotnet/AspNetCore.Docs/blob/f2e47cbf3bc88630ea4af98b4d07048f73cd8736/aspnetcore/blazor/hybrid/routing.md) | [ASP.NET Core Blazor Hybrid routing and navigation](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/routing?branch=pr-en-us-31231) |


<!-- PREVIEW-TABLE-END -->

BethMassi commented:
Thank you! I'll take a deeper look this week but couple comments for now..
- There's no plans at this time to add more support for deep linking in product, so the first bit can be removed
- We should mention the bit of code to set the blazorWebview.StartPath per https://github.com/dotnet/maui/issues/3788#issuecomment-1421550198

guardrex commented:
Very good ... I'll take care of both first thing Tuesday morning.

guardrex commented:
Start path coverage is already here. It's in the routing article at ...

https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/routing?view=aspnetcore-8.0&pivots=maui#get-or-set-a-path-for-initial-navigation

guardrex commented:
Latest updates ...

* Removed the section's introductory remarks on future work in this area.
* Added a short section at the end for setting the start path and cross-linked it to the start path section.
* Added the code-behind approach for setting the start path to the start path section.
